### DISCUSSION TITLE
nebnext-human-tcr-bcr-full-length preset not found for BCR and TCR analysis with NEB kit

### DISCUSSION THREAD

**Discussion by NoELJ22** [milaboratory/mixcr#1027] [Q&A] [answered]
Hello,

We are doing a BCR and TCR analysis on human data using the NEBNext® Immune Sequencing and I have a question about the available presets.

I saw in the presets proposed here https://mixcr.com/mixcr/reference/overview-built-in-presets/#new-england-biolabs the preset `nebnext-human-tcr-bcr-full-length`.
But when I launch the `mixcr listPresets` command with mixcr v4.2 I can't find the preset. We use mixcr v4.2 installed with conda.
Is the preset already available ?

As I couldn't use this preset I first ran the `nebnext-human-bcr-full-length` preset. At the exportClone step I got the IGH, IGK and IGL table, but also the TRA and TRB table
Same thing if I run the analysis with the `nebnext-human-tcr-full-length` preset. I get the TRA and TRB tables, but also IGH, IGK and IGL.
No matter which preset I choose, does mixcr align the Ig and Tr ?

Moreover if I compare the two results, there are differences even if they are very small.
I attach the reports for my 4 samples and the Chain usage reports, obtained with the BCR preset and the TCR preset:

[chainUsage_bcr_preset.pdf](https://github.com/milaboratory/mixcr/files/10909803/chainUsage_bcr_preset.pdf)
[chainUsage_tcr_preset.pdf](https://github.com/milaboratory/mixcr/files/10909804/chainUsage_tcr_preset.pdf)
[NEB1-1995-100BT_S5_bcr_preset.report.txt](https://github.com/milaboratory/mixcr/files/10909805/NEB1-1995-100BT_S5_bcr_preset.report.txt)
[NEB1-1995-100BT_S5_tcr_preset.report.txt](https://github.com/milaboratory/mixcr/files/10909806/NEB1-1995-100BT_S5_tcr_preset.report.txt)
[NEB2-5589-100BT_S6_bcr_preset.report.txt](https://github.com/milaboratory/mixcr/files/10909808/NEB2-5589-100BT_S6_bcr_preset.report.txt)
[NEB2-5589-100BT_S6_tcr_preset.report.txt](https://github.com/milaboratory/mixcr/files/10909809/NEB2-5589-100BT_S6_tcr_preset.report.txt)
[NEB3-5631-100BT_S7_bcr_preset.report.txt](https://github.com/milaboratory/mixcr/files/10909810/NEB3-5631-100BT_S7_bcr_preset.report.txt)
[NEB3-5631-100BT_S7_tcr_preset.report.txt](https://github.com/milaboratory/mixcr/files/10909811/NEB3-5631-100BT_S7_tcr_preset.report.txt)
[NEB4-5589-50BT_S8_bcr_preset.report.txt](https://github.com/milaboratory/mixcr/files/10909812/NEB4-5589-50BT_S8_bcr_preset.report.txt)
[NEB4-5589-50BT_S8_tcr_preset.report.txt](https://github.com/milaboratory/mixcr/files/10909813/NEB4-5589-50BT_S8_tcr_preset.report.txt)

Thank you for your help.

**Comment by mizraelson** [comment, marked as answer]
Hi,
The preset should be available in the last release:
https://github.com/milaboratory/mixcr/releases/

Yes, MiXCR aligns both Ig and TCR, but aligner algorithms are different. So the BCR aligner is more optimized for that type of data, and TCR aligner works faster, so its possible the outcome differs a bit if you analyze BCR data with TCR aligner for example.

The new preset which works for both TCR and BCR is optimized for both.

**Reply by NoELJ22** [reply]
Thank you for your answer.

I made a mistake in my previous post. I am using mixcr v4.2 (not V2.4), last release:
![mixcr_version](https://user-images.githubusercontent.com/115169629/223460799-962e346e-85a2-477f-955b-77bf9b8b7d70.png)
But i had an error when i used the `human-tcr-bcr-full-length preset`
![error_present_neb_tcr_bcr](https://user-images.githubusercontent.com/115169629/223460863-e9bdd942-dc2c-4400-b51c-d47c4e7b90a1.png)
And i can't find it in the preset list with `mixcr listPresets` command 
![mixcr_preset_existe](https://user-images.githubusercontent.com/115169629/223460872-34e4af62-d901-4c22-bb80-a1ce946fd184.png)

**Reply by mizraelson** [reply]
Sorry, my mistake. We have it in the develop version available here:

https://link.milaboratory.com/software/mixcr/mixcr-develop.zip

Later this week we will release an update and it will become available in the master version as well.

**Reply by NoELJ22** [reply]
Great, thank you for your help
